pkg/controller/vm_controller.go

```diff
@@ -338,6 +357,89 @@ func (r *VMReconciler) buildVMPod(ctx context.Context, vm *virtv1alpha1.VirtualM
 				Args:            []string{volumeMount.MountPath + "/rootfs.raw", strconv.FormatInt(volume.ContainerRootfs.Size.Value(), 10)},
 				VolumeMounts:    []corev1.VolumeMount{volumeMount},
 			})
+		case volume.PersistentVolumeClaim != nil:
+			vmPod.Spec.Volumes = append(vmPod.Spec.Volumes, corev1.Volume{
+				Name: volume.Name,
+				VolumeSource: corev1.VolumeSource{
+					PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+						ClaimName: volume.PersistentVolumeClaim.ClaimName,
+					},
+				},
+			})
+
+			pvcKey := types.NamespacedName{
+				Namespace: vm.Namespace,
+				Name:      volume.PersistentVolumeClaim.ClaimName,
+			}
+			var pvc corev1.PersistentVolumeClaim
+			if err := r.Client.Get(ctx, pvcKey, &pvc); err != nil {
+				return nil, fmt.Errorf("get volume PVC: %s", err)
+			}
+
+			volumeMode := corev1.PersistentVolumeFilesystem
+			if pvc.Spec.VolumeMode != nil {
+				volumeMode = *pvc.Spec.VolumeMode
+			}
+
+			if volumeMode == corev1.PersistentVolumeFilesystem {
+				// TODO: filesystem PVC is supported by CDI now, do we need support it directly?
+				return nil, fmt.Errorf("don't support filesystem PVC now: %s", volume.Name)
```

Comment:
We need to support FS PVC too

Reply:
support FS PVC by DataVolume is a better choice, otherwise would repeat a large chunk of codes in vintink.
for example DataVolume support configure a  [filesystem overhead](https://github.com/kubevirt/containerized-data-importer/blob/main/doc/cdi-config.md#options) for each storageClass. we can even remove Block PVC, and support it by DataVolume too.

Reply:
It would be an overkill to remove PVC support. We should give the options to use PVC directly without CDI when the "data" is already in the PV